lib/message/decrypt.js

```diff
@@ -1,34 +1,39 @@
-import { openpgp } from '../openpgp';
+import { isStream } from '@openpgp/web-stream-tools';
+import { decrypt, readSignature } from 'openpgp';
 import { decodeUtf8Base64, binaryStringToArray, arrayToBinaryString } from '../utils';
 import { serverTime } from '../serverTime';
 import { getMessage, handleVerificationResult } from './utils';
 import processMIME from './processMIME';
 import { getEncMessageFromEmailPM, getEncRandomKeyFromEmailPM } from './compat';
 import { VERIFICATION_STATUS } from '../constants';
+import { AES256 as AES_CFB } from '../crypto/cfb';
 
-export async function decryptMessage(options) {
-    const { publicKeys = [] } = options;
-    options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
+export async function decryptMessage({ date = serverTime(), encryptedSignature, armor = true, ...options }) {
+    const sanitizedOptions = { ...options, date };
+
+    if (!options.format) {
+        sanitizedOptions.format = armor ? 'armored' : 'object'; // TODO change, remove `armor` param?
+    }
 
     try {
         // If encryptedSignature exists, decrypt and use it
-        if (options.encryptedSignature) {
-            const decryptedSignature = await openpgp.decrypt({
-                ...options,
-                message: options.encryptedSignature,
+        if (encryptedSignature) {
+            const { data: decryptedSignature } = await decrypt({
+                ...sanitizedOptions,
+                message: encryptedSignature,
                 format: 'binary'
             });
-            options.signature = await openpgp.signature.read(decryptedSignature.data);
+            sanitizedOptions.signature = await readSignature({ binarySignature: decryptedSignature });
         }
 
-        const result = await openpgp.decrypt(options);
-        const verificationResult = handleVerificationResult(result, publicKeys, options.date);
+        const result = await decrypt(sanitizedOptions);
+        const verificationResult = handleVerificationResult(result);
 
         let verified = verificationResult.then((result) => result.verified);
         let verifiedSignatures = verificationResult.then((result) => result.signatures);
         let errors = verificationResult.then((result) => result.errors);
 
-        if (!openpgp.stream.isStream(result.data)) {
+        if (!isStream(result.data)) {
             verified = await verified;
             verifiedSignatures = await verifiedSignatures;
             errors = await errors;
```

Comment:
is `errors` still relevant?

Reply:
No, can be removed

Reply:
It might still make sense to keep it around, we'll remove in separate PR in case

Reply:
Sorry, I was also wrong, it's still needed I think as `handleVerificationResult` converts the `verified` promise to false and moves the error to this array

Reply:
Yes, but we might change the result data anyway with the worker